### ISSUE TITLE
`wr.s3.to_parquet` with `sanitize_columns=True` creates a side effect on the original dataframe

### ISSUE THREAD

**Issue by yuvalshi0** [aws/aws-sdk-pandas#1914]
### Describe the bug

I have a dataframe with an hyphen as the column name, when I try to upload the dataframe to s3 (in my case, to a glue table in a catalog), the column sanitising modifies the original dataframe, which causes an unwanted side effect.

### How to Reproduce

```python
import awswrangler as wr
import pandas as pd

df = pd.DataFrame({"foo-bar": [1, 2, 3]})
orig_columns = df.columns

wr.s3.to_parquet(
    df=df,
    path="s3://my-bucket/my-path",
    sanitize_columns=True,
)

assert df.columns == orig_columns  # raises AssertonError, because the column was renamed to foo_bar
```

### Expected behavior

* `wr.s3.to_parquet` shouldn't modify the original dataframe
* Allow to disable the `sanitize_columns`! (#533) 

### OS

linux

### Python version

3.10

### AWS SDK for pandas version

2.18

**Comment by malachi-constant** [comment]
The `sanitize_columns` behavior is enforced whenever writing to S3 with a glue db and table specified.

> [sanitize_columns (bool)](https://aws-sdk-pandas.readthedocs.io/en/stable/stubs/awswrangler.s3.to_parquet.html#awswrangler.s3.to_parquet) – True to sanitize columns names (using wr.catalog.sanitize_table_name and wr.catalog.sanitize_column_name) or False to keep it as is. True value behaviour is enforced if database and table arguments are passed.

The [Glue documentation](https://docs.aws.amazon.com/athena/latest/ug/glue-best-practices.html#schema-names) also indicates 
> When you create schema in AWS Glue to query in Athena, consider the following:
> - A database name cannot be longer than 255 characters.
> - A table name cannot be longer than 255 characters.
> - A column name cannot be longer than 255 characters.
> - The only acceptable characters for database names, table names, and column names are lowercase letters, numbers, and the underscore character.

Given you can still create a schema with hyphens (`-`) in column names it's possible we can consider an override for this behavior, but I'll let the rest of the team weigh in. @aws/aws-sdk-pandas

**Comment by yuvalshi0** [comment]
> The `sanitize_columns` behavior is enforced whenever writing to S3 with a glue db and table specified.
> 
> > [sanitize_columns (bool)](https://aws-sdk-pandas.readthedocs.io/en/stable/stubs/awswrangler.s3.to_parquet.html#awswrangler.s3.to_parquet) – True to sanitize columns names (using wr.catalog.sanitize_table_name and wr.catalog.sanitize_column_name) or False to keep it as is. True value behaviour is enforced if database and table arguments are passed.
> 
> The [Glue documentation](https://docs.aws.amazon.com/athena/latest/ug/glue-best-practices.html#schema-names) also indicates
> 
> > When you create schema in AWS Glue to query in Athena, consider the following:
> > 
> > * A database name cannot be longer than 255 characters.
> > * A table name cannot be longer than 255 characters.
> > * A column name cannot be longer than 255 characters.
> > * The only acceptable characters for database names, table names, and column names are lowercase letters, numbers, and the underscore character.
> 
> Given you can still create a schema with hyphens (`-`) in column names it's possible we can consider an override for this behavior, but I'll let the rest of the team weigh in. @aws/aws-sdk-pandas

Even with the fact that the `sanitize_columns` behavior was enforced (implicitly, in my glue table case), I don't think it is expected to modify the original dataframe which was passed to.

**Comment by malachi-constant** [comment]
Well if by modify you mean column names being changed that would be expected as that is what is happing during the [sanitization process](https://github.com/aws/aws-sdk-pandas/blob/bd518878c8f661091a143f4474ea97032e80eb37/awswrangler/catalog/_utils.py#L99).

**Comment by yuvalshi0** [comment]
> Well if by modify you mean column names being changed that would be expected as that is what is happing during the [sanitization process](https://github.com/aws/aws-sdk-pandas/blob/bd518878c8f661091a143f4474ea97032e80eb37/awswrangler/catalog/_utils.py#L99).

I don't think the wrangler should be changing the original dataframe, that can cause unexpected bugs, specially if other sections of your code are using the same dataframe, you can "protect" your dataframe by doing `df_ = df.copy()` before every call to the wrangler, but that's just a workaround

**Comment by malachi-constant** [comment]
An update has been merged to `branch/release-3.0.0`. This will be released as part of our next major version update as it is  considered a breaking change. #1919
